platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileReaderPbj.java

```diff
@@ -336,17 +344,38 @@ protected int leaseFileChannel() throws IOException {
         // Although openNewFileChannel() is thread safe, it makes sense to check the count here.
         // Since the channels are never closed (other than when the data file reader is closed),
         // it's safe to check count against MAX_FILE_CHANNELS
-        if ((inUse / count > THREADS_PER_FILECHANNEL) && (count < MAX_FILE_CHANNELS)) {
+        if (((float) inUse / count > THREADS_PER_FILECHANNEL) && (count < MAX_FILE_CHANNELS)) {
             openNewFileChannel(count);
             count = fileChannelsCount.get();
+            final int newChannelLease = count - 1;
+            fileChannelLeases.incrementAndGet(newChannelLease);
+            return newChannelLease;
+        }
+        // find a channel with the least number of leases
```

Comment:
The remainder of this method (except the two last lines) can be wrapped into "else", so there is only one "return" statement and one "incrementAndGet()" call

Reply:
Done